community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/ASTRewriter.scala

```diff
@@ -47,7 +47,7 @@ class ASTRewriter(rewriterSequencer: (String) => RewriterStepSequencer, shouldEx
       enableCondition(containsNoReturnAll),
       foldConstants,
       ApplyRewriter("extractParameters", extractParameters),
-      nameMatchPatternElements,
+      nameMatchAndMergePatternElements,
```

Comment:
There is already a nameUpdatingClauses Rewriter. That could be used here instead of changing nameMatchPatternElements